rmw/include/rmw/rmw.h

```diff
@@ -838,15 +838,66 @@ rmw_subscription_get_actual_qos(
   const rmw_subscription_t * subscription,
   rmw_qos_profile_t * qos);
 
-/// Take an incoming message from a subscription.
+/// Take an incoming ROS message.
 /**
- * Take an incoming ROS message from a given subscription.
+ * Read a ROS message received by the given subscription, removing it from internal queues.
```

Comment:
```suggestion
 * Take a ROS message received by the given subscription, removing it from internal queues.
```

Read means something else in the context of DDS, so it's better we don't mix these words up.

Reply:
Yeah, I took a look at [RTI's docs](https://community.rti.com/static/documentation/connext-dds/5.2.0/doc/manuals/connext_dds/html_files/RTI_ConnextDDS_CoreLibraries_UsersManual/Content/UsersManual/AccessingData_Samples_ReadTake.htm) for a good definition of `take` (to avoid recursive definitions) and it's essentially a destructive read. But I'm fine with keeping `take` too.

Reply:
It is, but in the future we will most likely have an `rmw_read()` as well and using the term `read` here is going to be confusing.